docs/proposals/groups.md

```diff
@@ -0,0 +1,94 @@
+
+
+## Use Cases
+ 1.  I want to enable a subset of users to be able to organize sets of Users into OrgGroups, but I only want those changes to affect policy evaluation across some projects.
+ 1.  I want to be able to define some ClusterGroups that will affect policy evaluation across all projects.
+
+
+## Overview
+Organizations are cluster-scoped resources that contains a set of OrgGroups, a list of admins (users that can manage the groups), and a list of members (users that can be added to the OrgGroups).  ClusterGroups are cluster-scoped sets of users that are managed by a cluster-admin.  OrgGroups are organization scoped sets of users that are managed by the Organization.AdminUsers.  Projects may have at most one owning Organization.  OrgGroups may only be referenced in RoleBindings of Projects owned by the Organization.
+
+## ClusterGroups
+ClusterGroups are cluster-scoped sets of users that honored for all projects in the cluster.  That means that if UserA is added to ClusterGroupB and ProjectC has a rolebinding that says "ClusterGroupB is an editor", UserA now has editor permissions in ProjectC.  ClusterGroups will be described using an API object that looks like this:
+```
+// ClusterGroup contains the set of users that belong to a given ClusterGroup.  The name may not contain ":"
+type ClusterGroup struct {
+	kapi.TypeMeta
+	kapi.ObjectMeta
+
+	// Users is a list of Users that are members of the ClusterGroup
+	Users util.StringSet
+}
+```
+
+A reverse lookup index will be built in memory (similar to the project cache) to be allow the authenticator to make fast lookups for ClusterGroup membership determination.
+
+
+## OrgGroups
+OrgGroups look a lot like ClusterGroups, but they are always scoped to a particular Organization and they behave differently.  OrgGroups are managed by Organization.Admins (ClusterGroups are managed by cluster admins), OrgGroups can only contain Users that are Organization.Members (ClusterGroups can contain any user), OrgGroups can only be bound to roles in projects owned by the Organization (ClusterGroups can be bound to roles in any project).  They will be cluster scoped resources, but we'll indicate the Organization scope of a OrgGroup by prefixing every OrgGroup name with Organization name.
+```
+// OrgGroup contains the set of Users that belong to a given OrgGroup.  The name of a OrgGroup is required to 
+// to be in the form of "OrganizationName:OrgGroupName".
+type OrgGroup struct {
+	kapi.TypeMeta
+	kapi.ObjectMeta
+
+	// Users is a list of Users that are members of the OrgGroup
+	Users util.StringSet
+}
+```
+
+A reverse lookup index will be built in memory (similar to the project cache) to be allow the authenticator to make fast lookups for OrgGroup membership determination.  A valid OrgGroup.Name can never be used as a ClusterGroup.Name and vice-versa.  This detail is important later on for authorization parity with kube and backwards compatibility with v3 of OpenShift.
```

Comment:
More detail as to why - prefixing the org name would be sufficient to preserve uniqueness.